src/components/checkbox/index.tsx

```diff
@@ -1,64 +1,98 @@
 import _ from 'lodash';
-import PropTypes from 'prop-types';
-import React from 'react';
-import {Animated, Easing, StyleSheet} from 'react-native';
+import React, {Component} from 'react';
+import {Animated, Easing, StyleSheet, StyleProp, TouchableOpacityProps, ViewStyle} from 'react-native';
 import {Colors} from '../../style';
+//@ts-ignore
 import Assets from '../../assets';
-import {BaseComponent} from '../../commons';
+import {asBaseComponent, BaseComponentInjectedProps, forwardRef} from '../../commons/new';
 import TouchableOpacity from '../touchableOpacity';
 
 const DEFAULT_SIZE = 24;
 const DEFAULT_COLOR = Colors.blue30;
 const DEFAULT_ICON_COLOR = Colors.white;
 const DEFAULT_DISABLED_COLOR = Colors.dark70;
 
+interface CheckboxProps {
+  /**
+   * The value of the Checkbox. If true the switch will be turned on. Default value is false.
+   */
+  value?: boolean;
+  /**
+   * Invoked with the new value when the value changes.
+   */
+  onValueChange?: Function;
+  /**
+   * Whether the checkbox should be disabled
+   */
+  disabled?: boolean;
+  /**
+   * The Checkbox color
+   */
+  color?: string;
+  /**
+   * The size of the checkbox. affect both width and height
+   */
+  size?: number;
+  /**
+   * The Checkbox border radius
+   */
+  borderRadius?: number;
+  /**
+   * The icon asset to use for the selected indication (accept only local assets)
+   */
+  selectedIcon?: number;
+  /**
+   * The selected icon color
+   */
+  iconColor?: string;
+  /**
+   * Additional styling
+   */
+  style?: StyleProp<ViewStyle>
```

Comment:
Didn't see any other ts including this type but without it example screen shouts:
`Type '{ value: boolean; onValueChange: (value2: any) => void; borderRadius: number; size: number; color: string; selectedIcon: number; style: { marginBottom: number; }; }' is not assignable to type 'IntrinsicAttributes & CheckboxProps & { children?: ReactNode; }'.
  Property 'style' does not exist on type 'IntrinsicAttributes & CheckboxProps & { children?: ReactNode; }'.ts(2322)`

Is there another solution for this?